### PR TITLE
Move random hex generator function to the `src/shared` folder

### PR DESCRIPTION
**Pull request by esanzgar**
Code in the `src/annotator` can and will use the `generateHexString` functionality.

I also cleanup some unusual code.

### CODE REVIEW

File: src/sidebar/util/postmessage-json-rpc.js [hunk 1]
```diff
@@ -33,7 +28,7 @@ export function call(
   params = [],
   timeout = 2000,
   window_ = window,
-  id = generateId()
+  id = generateHexString(10)
```
Comment (robertknight) [inline src/sidebar/util/postmessage-json-rpc.js:31]: I think we can just suppress the code coverage complaint here.

**Review by robertknight — APPROVED** [review]
LGTM

File: src/sidebar/util/postmessage-json-rpc.js (hunk 1 above)
Comment (esanzgar) [reply src/sidebar/util/postmessage-json-rpc.js]: I have removed the unnecessary test seam in a follow up commit.

File: src/annotator/annotation-sync.js [hunk 2]
```diff
@@ -108,7 +110,7 @@ export class AnnotationSync {
     if (annotation.$tag) {
       return annotation;
     }
-    tag = tag || window.btoa(Math.random().toString());
+    tag = tag || generateHexString(8);
```
Comment (esanzgar) [inline src/annotator/annotation-sync.js:113]: The only question I have is about this line.

This creates around 24 character long string of random alpha-numerical characters. This was different from `src/sidebar/services/annotations.js` that use `generateHexString(8)`.

Is there any value on having a different format for the `$tag` here than the one generated in the sidebar?

**Pull request merged**